### ISSUE TITLE
NumberFormatException instead of `Infinity` or `-Infinity`

### ISSUE THREAD

**Issue by catap** [scala-native/scala-native#1836]
Let assume some very small (or very big) number as string:
```
scala>     val a = "-274672389457236457826542634627345697228374687236476867674746" +
     |       "2342342342342342342342323423423423423423426767456345745293762384756" +
     |       "2384756345634568456345689345683475863465786485764785684564576348756" +
     |       "7384567845678658734587364576745683475674576345786348576847567846578" +
     |       "3456702897830296720476846578634576384567845678346573465786457863"
val a: String = -2746723894572364578265426346273456972283746872364768676747462342342342342342342342323423423423423423426767456345745293762384756238475634563456845634568934568347586346578648576478568456457634875673845678456786587345873645767456834756745763457863485768475678465783456702897830296720476846578634576384567845678346573465786457863
```
and try to convert to string like that:
```
scala> a.toDouble
val res0: Double = -Infinity
```

This is expected behaviour. But scala-native throw an exception like:
```
[info]   java.lang.NumberFormatException: -2746723894572364578265426346273456972283746872364768676747462342342342342342342342323423423423423423426767456345745293762384756238475634563456845634568934568347586346578648576478568456457634875673845678456786587345873645767456834756745763457863485768475678465783456702897830296720476846578634576384567845678346573465786457863
[info]   at java.lang.Throwable.fillInStackTrace(Unknown Source)
[info]   at java.lang.Double$$anonfun$parseDouble$1.apply(Unknown Source)
[info]   at java.lang.Double$$anonfun$parseDouble$1.apply(Unknown Source)
[info]   at scala.scalanative.unsafe.Zone$.apply(Unknown Source)
[info]   at java.lang.Double$.parseDouble(Unknown Source)
```

**Comment by LeeTibbert** [comment]
@catap  Thank you for reporting this defect and sorry that it took your time. The report is quite clear
and helpful. It helped me locate the code in the current mainline.  When I saw it, I had the distinct
feeling "Hmmm, I thought I had fixed that wonky errno check." and when hunting.

I believe that pending PR #1837 fixes this and a few other problems. If this Issue is making your
work difficult, we can correspond via private gitter about how to proceed.  A lot depends upon 
which other libraries you are using.

**Comment by catap** [comment]
@LeeTibbert I believe that this is different issue because my example doesn’t touch BigInteger implementation.

Just a Double.

**Comment by LeeTibbert** [comment]
I bungled the Issue number once again.  Fixed now.  I must be doing something wrong with cut & paste. Arrgh!
I believe the issue is/was in Double.parseDouble().  

The PR is still open, so I can see if something similar to the problem you report is present as a test case.

**Comment by catap** [comment]
@LeeTibbert float ;)

**Comment by LeeTibbert** [comment]
These are the relevant cases in the pending PR.  

I can add the reproducer you kindly supplied and/or any other cases you
think wise or relevant.

```
  // Out of IEE754 range handling                                             

    //   Too big - java.lang.Double.MAX_VALUE times 10                          
    assert(Double.parseDouble("1.7976931348623157E309") ==
             Double.POSITIVE_INFINITY,
           "a20")

    //   Too big - Negative java.lang.Double.MAX_VALUE times 10                 
    assert(Double.parseDouble("-1.7976931348623157E309") ==
             Double.NEGATIVE_INFINITY,
           "a21")

    //   Too close to 0 - java.lang.Double.MIN_VALUE divided by 10              
    assert(Double.parseDouble("4.9E-325") == 0.0, "a22")
```

**Comment by catap** [comment]
@LeeTibbert yes, this is fixed this case.

BTW maybe to release a M3 version with a lot of small fixes Like this and without support Scala-2.13 Not such bad idea?

**Comment by LeeTibbert** [comment]
The Float case should be fixed also.  I was lazy and did not add a test case to FloatSuite.scala.
Perhaps I am relying too much on knowing that Float.toFloat exercises the same code path.
More likely just laziness: Too many bugs, too little time.

I can add that case by the light of day if you think it wise or proper. No problem.

**Comment by LeeTibbert** [comment]
English is a slippery language. By `The Float case should be fixed also. ` I meant "I believe is fixed, to some degree of certainty" rather than "Someday somebody should do that".

So that the community does not think me exceedingly lazy, I ported the Double test case, mutatis mutandis, to Float.
To exercise a slightly different path in the underlying code, I flipped the sign of the test.  All tests pass.

**Comment by catap** [comment]
@LeeTibbert I understand that and base ob my experiences I suggest to cover float also by the same test.

Why? Because in couple of years someone may re-introduce this bug as part of optimization and introduce it only to `float` made things much worst.

**Comment by sjrd** [comment]
I believe this was fixed in 65a168dba3c4c58d2e598616011d4562d65fe868 as well.

**Comment by LeeTibbert** [comment]
@catap  For exactly the reasons you mentioned, I added the test case for Double.parseDouble in DoubleSuite.scala
to FloatSuite.scala. There were two changes. First, the necessary one to change from Double.parseDouble to 
Float.parseFloat. The second was to flip the sign from negative to positive.  

The latter change is, granted, somewhat of a flourish. That is, not strictly necessary.    I did it so that
each of positive & negative infinity are tested,  just as 2 tests, not 2x2.

The two cases of "too close to zero", approaching from less than zero and greater than zero are not
covered.  If you think it well advised, I can review IEEE negative zero and add those cases.  Belts, suspenders,
and horse nails.
